backend/danswer/connectors/notion/connector.py

```diff
@@ -0,0 +1,222 @@
+"""Notion reader."""
+import time
+from dataclasses import dataclass, fields
+from typing import Any, Dict, List, Optional
+
+import requests
+
+from danswer.configs.app_configs import INDEX_BATCH_SIZE
+from danswer.configs.constants import DocumentSource
+from danswer.connectors.interfaces import GenerateDocumentsOutput
+from danswer.connectors.interfaces import LoadConnector
+from danswer.connectors.interfaces import PollConnector
+from danswer.connectors.interfaces import SecondsSinceUnixEpoch
+from danswer.connectors.models import Document
+from danswer.connectors.models import Section
+
+
+@dataclass
+class NotionPage:
+    """Represents a Notion Page object"""
+    id: str
+    created_time: str
+    last_edited_time: str
+    archived: bool
+    properties: Dict[str, Any]
+    url: str
+
+    def __init__(self, **kwargs):
+        names = set([f.name for f in fields(self)])
+        for k, v in kwargs.items():
+            if k in names:
+                setattr(self, k, v)
+
+
+@dataclass
+class NotionSearchResponse:
+    """Represents the response from the Notion Search API"""
+    results: List[Dict[str, Any]]
+    next_cursor: Optional[str]
+    has_more: bool = False
+
+    def __init__(self, **kwargs):
+        names = set([f.name for f in fields(self)])
+        for k, v in kwargs.items():
+            if k in names:
+                setattr(self, k, v)
+
+
+# TODO - Add the ability to optionally limit to specific Notion databases
+class NotionConnector(LoadConnector, PollConnector):
+    """Notion Page connector that reads all Notion pages
+    this integration has been granted access to.
+
+    Arguments:
+        batch_size (int): Number of objects to index in a batch
+    """
+
+    def __init__(self, batch_size: int = INDEX_BATCH_SIZE) -> None:
```

Comment:
Personal preference, not sure what you think. Generally I like placing init on top then, private functions, then public functions. But this is completely up to you, this is fine as is.